### PR TITLE
Fix BLK_SHIFT may be wrong in some platforms #128

### PR DESCRIPTION
**Pull request by fengyichui**
Fix BLK_SHIFT may be wrong in some platforms

### CODE REVIEW

File: src/nnn.c [hunk 1]
```diff
@@ -417,6 +417,18 @@ static uchar crc8fast(uchar const message[], size_t n)
 	return remainder;
 }
 
+/* Get platform block shift */
+static int get_blk_shift(void)
```
Comment (jarun) [inline src/nnn.c]: Please use `ffs()` or the appropriate one from the family (`man ffs`) based on the size required.

Comment (jarun) [reply src/nnn.c]: We already use it in the code.

Comment (fengyichui) [reply src/nnn.c]: oh, it's best. 👍

Comment (fengyichui) [reply src/nnn.c]: I want to use `ctz` to do ‘Count Tailing Zeros’, but I can't find the API.

File: src/nnn.c [hunk 2]
```diff
@@ -3399,6 +3399,9 @@ int main(int argc, char *argv[])
 	char *ipath = NULL;
 	int opt;
 
+	// Get platform block shift
+	BLK_SHIFT = ffs(S_BLKSIZE) - 1;
```
[the remark this answers is not in the document]
Comment (fengyichui) [reply src/nnn.c]: But `-S` option need  to initialize it.

[the remark this answers is not in the document]
Comment (fengyichui) [reply src/nnn.c]: Put it into `-S` case ?

**Comment by jarun** [comment]
Thank you!

**Comment by jarun** [comment]
@fengyichui unfortunately this breaks on Termux where I couldn't find `S_BLKSIZE` defined in `sys/stat.h`.

I have raised https://github.com/termux/termux-app/issues/925.

**Comment by fengyichui** [comment]
@jarun We can add the following code to the front of `nnn.c`.

```c
#ifndef S_BLKSIZE
#define S_BLKSIZE 512
#endif
```

**Comment by fengyichui** [comment]
@jarun We can refer to the source code of [ncdu](https://dev.yorhel.nl/ncdu).

**Comment by jarun** [comment]
I have already added this at https://github.com/jarun/nnn/commit/362fd8ce78abe0115788b327eb16f4f9751506e7.

**Pull request merged**